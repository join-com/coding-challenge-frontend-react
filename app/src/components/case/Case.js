import React from 'react';
import './Case.css';

export default class Case extends React.Component {
  render() {
    return (
      <section className="card mb-3">
        <section className="row no-gutters">
          <section className="col-md-3">
            <img src={this.props.imageSrc} className="card-img" />
          </section>
          <section className="col-md-9">
            <section className="card-body">
              <h5 className="card-title">{this.props.title}</h5>
              <p className="card-text">{this.props.description}</p>
              <p className="card-text">
                <strong>Date of theft: </strong>
                {this.props.dateOfTheft}
              </p>
              <p className="card-text">
                <strong>Reported: </strong>
                {this.props.reported}
              </p>
              <p className="card-text">
                <strong>Location: </strong>
                {this.props.location}
              </p>
            </section>
          </section>
        </section>
      </section>
    )
  }
}
