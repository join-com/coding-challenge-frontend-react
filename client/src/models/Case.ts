interface IMedia {
  image_url: string;
  image_thumb: string;
}
interface ISource {
  name: string;
  html_url: string;
}
interface Case {
  id: number;
  address: string; 
  title: string;
  description: string;
  occured_at: number;
  updated_at: number;
  url: string;
  location_description: string;
  location_type: string; 
  media: IMedia;
  source: ISource;
}

export default Case;
