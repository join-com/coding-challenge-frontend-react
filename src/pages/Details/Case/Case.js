import React from 'react'
import styled from 'styled-components'

const Title = styled.h1`
  font-weight: 700;
  font-size: 30px;
`

const Description = styled.h2`
  text-transform: uppercase;
  font-weight: 400;
  font-size: 20px;
`

export default function Case({ item }) {
  return (
    <div>
      <Title>{item.title}</Title>
      <p>
        Stolen {item.occurred_at} at {item.address}
      </p>
      <div>Map</div> {/* TODO implement the map */}
      <Description>Description of incident</Description>
      <p>{item.description}</p>
    </div>
  )
}
