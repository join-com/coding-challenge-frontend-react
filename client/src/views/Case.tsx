import React, { EffectCallback, FC, Fragment, useEffect, useState } from "react";
import { match as IMatch } from "react-router-dom";
import Styled from "styled-components";
import Map from "../components/Map";
import Spinner from "../components/Spinner";
import { fetchCaseDetails, fetchSourceDetails} from "../helpers/Api";
import { FormatUTCDate } from "../helpers/Formatters";
import Case from "../models/ICase";

interface ICaseMatch {
  id: string;
}
interface ICaseDetailsProps {
    className?: string;
    match: IMatch<ICaseMatch>;
}

const CaseDetails: FC<ICaseDetailsProps> = ({ className, match }) => {
   const [caseData, setCaseData] = useState({} as Case);
   const [loading, setLoading] = useState(true);
   useEffect(() => {
    const doFetchAllCases = async () => {
        const result = await fetchCaseDetails(match.params.id);
        setCaseData(result);
        setLoading(false);
      };
    doFetchAllCases();
   }, []);
   return <div className={className}>
    {!loading && <Fragment>
      <h2> {caseData.title} </h2>
    <section>
        <p>
            Stolen {FormatUTCDate(caseData.occurred_at)}
            <span className="signify">at</span>
            <span className="address">{caseData.address}</span>
       </p>
    </section>
    <section>
        {caseData.feature && <Map coordinates={caseData.feature.geometry.coordinates}/>}
    </section>
    <h2>
      Description of incident
    </h2>
    <p>{caseData.description}</p>
    </Fragment>}
    {loading && <Spinner />}
</div>;
};

export default Styled(CaseDetails)`
  padding: 10px;
  h2 {
    font-size: 1.8em;
  }
  .signify {
    font-weight: 800;
    margin: 5px;
  }
`;
