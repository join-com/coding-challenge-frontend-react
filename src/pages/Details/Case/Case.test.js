import React from 'react'
import { shallow } from 'enzyme'
import Case from './Case'

it('renders the Case', () => {
  const item = {
    title: 'title',
    description: 'description',
    occurred_at: 12214324324,
    address: 'Halle, 06120, DE',
  }
  const wrapper = shallow(<Case item={item} />)
  expect(wrapper).toMatchSnapshot()
})
