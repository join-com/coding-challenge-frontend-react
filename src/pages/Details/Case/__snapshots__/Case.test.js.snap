// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Case 1`] = `
<div>
  <styled.h1>
    title
  </styled.h1>
  <p>
    Stolen 
    12214324324
     at 
    Halle, 06120, DE
  </p>
  <div>
    Map
  </div>
   
  <styled.h2>
    Description of incident
  </styled.h2>
  <p>
    description
  </p>
</div>
`;
